src/Polynomials.jl

```diff
@@ -60,22 +58,22 @@ p + q                  # ERROR: Polynomials must have same variable.
 ```
 
 """
-immutable Poly{T<:Number}
-    a::Vector{T}
-    var::Symbol
-    function Poly(a::Vector{T}, var)
-        # if a == [] we replace it with a = [0]
-        if length(a) == 0
-            return new(zeros(T,1), @compat Symbol(var))
-        else
-            # determine the last nonzero element and truncate a accordingly
-            a_last = max(1,findlast(x->x!=zero(T), a))
-            new(a[1:a_last], @compat Symbol(var))
-        end
+immutable Poly{T}
+  a::Vector{T}
+  var::Symbol
+  @compat function (::Type{Poly}){T<:Number}(a::Vector{T}, var::SymbolLike = :x)
+    # if a == [] we replace it with a = [0]
+    if length(a) == 0
+      return new{T}(zeros(T,1), @compat Symbol(var))
+    else
+      # determine the last nonzero element and truncate a accordingly
+      a_last = max(1,findlast(x->x!=zero(T), a))
+      new{T}(a[1:a_last], @compat Symbol(var))
     end
+  end
 end
 
-@compat Poly{T<:Number}(a::Vector{T}, var::Union{AbstractString,Symbol,Char}=:x) = Poly{T}(a, var)
+Poly(n::Number, var::SymbolLike = :x) = Poly([n], var)
```

Comment:
I am not sure if one would need `Poly(n::Number)` but if I am to obtain a `Poly(n::Number)`, still I need to write it in vector notation. For this reason, I just provided that for convenience.

When it comes to relaxing the `T<:Number` constraint, we should be also careful --- as of now, the way I have defined the inner constructor, `Poly` objects are not constructible from objects other than `Number`. That's why I relaxed the definition to `immutable Poly{T}` while keeping the constraint (implicitly) in the inner constructor. If that restriction is going to be relaxed, we need to remove the restriction in the inner constructor definition, too.